File: server/php/src/Application/Handlers/TokenPaymentHandler.php
<?php

namespace OnlinePayments\ExampleApp\Application\Handlers;

use OnlinePayments\ExampleApp\Application\Domain\Enums\Payments\PaymentMethodType;
use OnlinePayments\ExampleApp\Application\DTOs\Payment\RequestDto;
use OnlinePayments\ExampleApp\Application\DTOs\Payment\ResponseDto;
use OnlinePayments\ExampleApp\Application\Interfaces\Handlers\PaymentMethodHandlerInterface;
use OnlinePayments\ExampleApp\Application\Interfaces\SDKClients\PaymentClientInterface;

final readonly class TokenPaymentHandler implements PaymentMethodHandlerInterface
{
    public function __construct(
        private PaymentClientInterface $paymentClient
    ) {}

    public function getSupportedMethod(): PaymentMethodType
    {
        return PaymentMethodType::TOKEN;
    }

    /**
     * @throws \Exception
     */
    public function handle(RequestDto $requestDto): ResponseDto
    {
        return $this->paymentClient->createPayment($requestDto);
    }
}

File: server/python/app/application/handlers/token_payment_handler.py
from app.application.domain.enums.payments.payment_method_type import PaymentMethodType
from app.application.dtos.create_payment.request_dto import CreatePaymentRequestDto
from app.application.interfaces.handlers.payment_handler_interface import PaymentHandlerInterface

class TokenPaymentHandler(PaymentHandlerInterface):

    def __init__(self, payment_client):
        self._payment_client = payment_client

    def supported_method(self) -> PaymentMethodType:
        return PaymentMethodType.TOKEN

    def handle(self, request_dto: CreatePaymentRequestDto):
        return self._payment_client.create_payment(request_dto)